native/Avalonia.Native/src/OSX/WindowImpl.mm

```diff
@@ -517,7 +521,7 @@
                         ExitFullScreenMode();
                     }
 
-                    if (IsZoomed()) {
+                    if (IsZoomed() && shouldResize == true) {
```

Comment:
Shouldn't the whole `switch` be skipped if `shouldResize == false`? The `SetWindowState(Maximized, false)` call might move the window otherwise (it's probably a no-op currently though).

Reply:
Updated with https://github.com/AvaloniaUI/Avalonia/pull/19088/commits/921ca488e959d41831a9d886f6510ebd0dd39ed1

You're right, it would make more sense to wrap that around the switch itself.